internal/api/auth/basic_auth.go

```diff
@@ -12,6 +13,44 @@ const (
 	realm                = `Basic realm="restricted"`
 )
 
+type authMode int
+
+const (
+	NoAuth authMode = iota
+	LDAPMode
+)
+
+type BasicAuth struct {
+	ldapAuth *LDAPAuth
+	mode     authMode
+}
+
+func NewBasicAuth() BasicAuth { return BasicAuth{mode: NoAuth} }
+
+func (b *BasicAuth) SetMode(mode authMode) { b.mode = mode }
+
+func (b *BasicAuth) ConfigureLdap(ldap *LDAPAuth) error {
+	if ldap == nil {
+		return errors.New("LDAP configuration is missing")
+	}
+	b.ldapAuth = ldap
+
+	return nil
+}
+
+func (b *BasicAuth) Wrap(next httprouter.Handle) httprouter.Handle {
+	switch b.mode {
+	case NoAuth:
+		return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) { next(w, r, ps) }
+	case LDAPMode:
+		return BasicAuthLDAP(b.ldapAuth, next)
+	default:
+		return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+			http.Error(w, "authentication issue: bad server configuration", http.StatusInternalServerError)
```

Comment:
I would also add logging in case of any `500` error that we encounter

Reply:
Done